### PR TITLE
Add `rayleigh` distribution to `random.py`

### PR DESCRIPTION
**Pull request by JiaYaobo**
Add a counterpart  of `numpy.random.rayleigh` for JAX

### CODE REVIEW

File: jax/_src/random.py
Comment (sharadmv) [inline jax/_src/random.py:1836]: Please fix indentation here.

**Review by sharadmv — APPROVED** [review]
LGTM other than indentation issue

File: jax/_src/random.py
Comment (JiaYaobo) [reply jax/_src/random.py]: Thanks for your review, modification is committed

File: jax/_src/random.py
Comment (JiaYaobo) [reply jax/_src/random.py:1871]: Thanks for your review!

**Review by jakevdp — APPROVED** [review]

**Comment by JiaYaobo** [comment]
@jakevdp Hi, the workflow indicates the check `import/copybara: Pending —Waiting for internal safe review approval`, do I need to do something else to activate this check?

**Comment by sharadmv** [comment]
Sorry, I can take care of that!

**Pull request merged**
